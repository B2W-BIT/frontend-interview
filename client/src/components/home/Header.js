import React, { Component, PropTypes } from 'react';

const propTypes = {
  coverPhoto: PropTypes.string,
  bgColor: PropTypes.string,
  stats: PropTypes.object
};

const defaultProps = {
  coverPhoto: '',
  bgColor: '',
  stats: {
    tweetCount: 0,
    followingCount: 0,
    followersCount: 0,
    likesCount: 0
  }
};

export default class Header extends Component {
  render() {
    const { coverPhoto, bgColor, stats } = this.props;

    return (
      <header style={{backgroundColor:'#'+bgColor, backgroundImage:'url('+coverPhoto+')'}}>
        <div id="header-menu-container">
          <div className="header-menu">
            <div className="header-menu-item">
              <span className="header-menu-item-title">TWEETS</span>
              <span className="header-menu-item-content">{stats.tweetCount}</span>
            </div>
            <div className="header-menu-item">
              <span className="header-menu-item-title">FOLLOWING</span>
              <span className="header-menu-item-content">{stats.followingCount}</span>
            </div>
            <div className="header-menu-item">
              <span className="header-menu-item-title">FOLLOWERS</span>
              <span className="header-menu-item-content">{stats.followersCount}</span>
            </div>
            <div className="header-menu-item">
              <span className="header-menu-item-title">LIKES</span>
              <span className="header-menu-item-content">{stats.likesCount}</span>
            </div>
          </div>
        </div>
      </header>
    );
  }
}

Header.propTypes = propTypes;
Header.defaultProps = defaultProps;
