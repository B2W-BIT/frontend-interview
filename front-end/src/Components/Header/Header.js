import React, {Component} from 'react'
import img from '../../../public/assets/images/capa.jpg'
import '../../../public/assets/css/style.css'
import NavBottonHeader from '../Header/NavBotton'
import TabContent from '../Timeline/TabContent';
import TabLi from '../Timeline/TabLi'
import Post from '../Timeline/Post'
import {connect} from 'react-redux'
import {fotoCapa} from '../../Action/Timeline'
import { forEach } from "lodash"

class Header extends Component {
    
    constructor(props) {
        super(props);
        this.state = {
            ativoAtual: 'tweets',
            ativoLi: 'tweets',
            fotoCapa: []
        }

        this.handleClick = this.handleClick.bind(this);
        this.fotoCapa();
    }

    handleClick(e) {
        e.preventDefault();
        //console.log(e.target.id);
        this.setState({ ativoAtual: e.target.id });
        this.setState({ ativoLi: e.target.id });
    }

    fotoCapa(){
        this.props.fotoCapa().then((success) => {
            var fotoCapa = this.state.fotoCapa;
            this.setState({ fotoCapa: success.data });
        });
    }

    render () {
        return (
            <div>
                <div className="top-header">
                    <div className="top-book">
                        <img src={this.state.fotoCapa}/>
                    </div>
                </div>
                <NavBottonHeader />

                <div className="tabs-content">
                    <ul>
                        <TabLi clicado={this.state.ativoLi == "tweets"} Content={<li className="tt-li" id="tweets" onClick={this.handleClick}>Tweets e respostas</li>} />
                        <TabLi clicado={this.state.ativoLi == "midia"} Content={<li className="tt-li" id="midia" onClick={this.handleClick}>Mídia</li>} />
                    </ul>
                 </div>

                 <TabContent ativo={this.state.ativoAtual == "tweets"} Content={<Post  />} />
                 <TabContent ativo={this.state.ativoAtual == "midia"} Content={"Mídia"} />
            </div>
        )
    }
}

Header.propTypes = {
    fotoCapa: React.PropTypes.func.isRequired,
}

export default connect(null, {fotoCapa})(Header);