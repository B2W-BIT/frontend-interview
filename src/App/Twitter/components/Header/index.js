import React, { Component } from 'react';
import Nav from './components/Nav';

class Header extends Component {
    constructor(props) {
        super(props)
        this.state = {}
    }

    componentDidMount() {
        fetch('http://localhost:8090/api?call=users/profile_banner')
            .then(d => d.json())
            .then(d => {
                this.setState({
                    twitterData: d
                })
            })
    }

    render() {
        if (!this.state.twitterData) return <p>Loading</p>

        let header = {
            marginBottom: '16px'
        };

        let headerInner = {
            // height: '280px',
            overflow: 'hidden',
            boxShadow: '0 1px 1px rgba(0,0,0,0.2)'
        };

        let headerImg = {
            // marginTop: '-100px',
            width: '100%'
        };

        return (
            <div style={header}>
                <div style={headerInner}>
                    <img style={headerImg} src={this.state.twitterData.sizes['1500x500'].url} />
                </div>
                <Nav />
            </div>

        );
    }
}

export default Header;