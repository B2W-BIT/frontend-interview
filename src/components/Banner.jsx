import React from 'react';
import { Component } from 'react';

export default class Banner extends Component {
	static propTypes = {
		src: React.PropTypes.string,
		alt: React.PropTypes.string
	}

	static defaultProps = {
		src: '',
		alt: ''
	}

	shouldComponentUpdate(nextProps) {
		return nextProps.src !== this.props.src;
	}

	render() {
		return (<img className="img-responsive" src={ this.props.src } alt={ this.props.alt } />);
	}
}