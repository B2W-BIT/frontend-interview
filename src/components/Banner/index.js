import React, { Component } from 'react'
import Navbar from 'Components/ProfileNavbar'

class Banner extends Component {
  componentDidMount() {
    this.props.fetchUserData()
  }

  render() {
    const {
      profile_banner_url
      , profile_image_400
      , statuses_count
      , friends_count
      , followers_count
      , favourites_count
    } = this.props.user.info

    const navBarProps = {
      statuses_count
      , friends_count
      , followers_count
      , favourites_count
    }

    const bannerStyle = {
      backgroundImage: `url(${profile_banner_url})`
    }

    return (
      <div>
        <div className="header">
          <div className="before" style={bannerStyle}></div>
        </div>
        <div className="container-fluid">
          <div className="row">
            <Navbar {...navBarProps} />
          </div>
        </div>
        <div className="container">
          <div className="profile-image-container">
            <div className="profile-image">
              <img src={profile_image_400} alt=""/>
            </div>
          </div>
        </div>
      </div>
    )
  }
}

export default Banner
