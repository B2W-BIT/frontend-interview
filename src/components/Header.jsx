import React from 'react';
import { Component } from 'react';

export default class Header extends Component {
	static propTypes = {
		tweets: React.PropTypes.number,
		following: React.PropTypes.number,
		followers: React.PropTypes.number,
		likes: React.PropTypes.number
	}

	static defaultProps = {
		tweets: 0,
		following: 0,
		followers: 0,
		likes: 0
	}

	shouldComponentUpdate(nextProps) {
		if(typeof this.props.tweets !== 'number' || typeof this.props.following !== 'number' || typeof this.props.followers !== 'number' || typeof this.props.likes !== 'number')
			return true;

		return this.props.tweets !== nextProps.tweets ||
			this.props.following !== nextProps.following ||
			this.props.followers !== nextProps.followers ||
			this.props.likes !== nextProps.likes;
	}

	render() {
		return (<div className="header-info">
			<div className="container">
				<div className="row">
					<div className="col-sm-offset-3 col-md-offset-3 col-lg-offset-3 col-xs-6 col-sm-2 col-md-2 col-lg-1">Tweets <br /> { this.props.tweets.toLocaleString('pt-BR') }</div>
					<div className="col-xs-6 col-sm-2 col-md-2 col-lg-1">Seguindo <br /> { this.props.following.toLocaleString('pt-BR') }</div>
					<div className="col-xs-6 col-sm-2 col-md-2 col-lg-1">Seguidores <br /> { this.props.followers.toLocaleString('pt-BR') }</div>
					<div className="col-xs-6 col-sm-2 col-md-2 col-lg-1">Curtidas <br /> { this.props.likes.toLocaleString('pt-BR') }</div>
				</div>
			</div>
		</div>);
	}
}