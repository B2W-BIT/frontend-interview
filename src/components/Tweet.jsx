import React from 'react';
import { Component } from 'react';
import TwitterUser from './TwitterUser.jsx';
import moment from 'moment';
import entities from '../utils/entities.js';

export default class Tweet extends Component {
	static contextTypes = {
		linkColor: React.PropTypes.string
	}

	renderAnswering() {
		if(this.props.in_reply_to_screen_name && this.props.entities.user_mentions && this.props.entities.user_mentions.length) {
			return (<div className="col-xs-offset-1 col-xs-11">
				Em resposta a { this.props.entities.user_mentions[0].name }
			</div>);
		}
		return null;
	}

	render() {
		return (<div className="tweet-block row">
			{ this.renderAnswering() }
			<div className="col-xs-1 tweet-img">
				<img className="img-responsive" src={ this.props.user.profile_image_url_https } alt="" />
			</div>
			<div className="col-xs-11">
				<TwitterUser verified={ this.props.user.verified } name={ this.props.user.name } screen_name={ this.props.user.screen_name } />
				- { moment(new Date(this.props.created_at)).format('DD [de] MMMM') }
				<br />
				{ entities(this.props.text, this.props.entities, this.context.linkColor, {
					src: 'user_mentions',
					url: 'screen_name',
					_url: 'http://twitter.com/',
					_from: 'screen_name',
					to: 'name'
				}, 'media', {
					src: 'urls',
					list: 'urls',
					url: 'expanded_url',
					_from: 'url',
					to: 'display_url'
				}, {
					src: 'url',
					list: 'urls',
					url: 'expanded_url',
					_from: 'url',
					to: 'display_url'
				}) }
			</div>
		</div>);
	}
}