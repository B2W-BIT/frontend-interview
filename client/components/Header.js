import io from 'socket.io-client';
import React from 'react';
import Tweet from './Tweet';



export default class Header extends React.Component {

	 render() {
    	return (
    		<div>
    			<div className="top-user row">	
    				<div className="col-lg-4 col-md-4 col-sm-4 col-xs-4">
	    				<ul className="main-list">
	    					<a href="#"><li>Home</li></a>
	    					<a href="#"><li>Moments</li></a>
	    					<a href="#"><li>Notifications</li></a>
	    					<a href="#"><li>Messages</li></a>
	    				</ul>
    				</div>
    				<div className="col-lg-2 col-md-2 col-sm-2 col-xs-2 col-lg-offset-2 col-md-offset-2 col-sm-offset-2 col-xs-offset-2">
    					<a href="#">
    						<img className="twitter-img" src="http://www.freeiconspng.com/uploads/twitter-icon-download-18.png" />
    					</a>
    				</div>
    				<div className="col-lg-2 col-md2 col-sm-2 col-xs-4">
						<div className="form-group has-feedback-left">
						    <input type="text" className="form-control" placeholder="Search Twitter" />
						    <i className="glyphicon glyphicon-search form-control-feedback"></i>
						</div>
    				</div>
    				<div className="col-lg-2 col-md2 col-sm-2 hidden-xs">
    					<img className="img-top-user-1" src="../images/img-top-user-1.png" />
    					<img className="img-top-user-2" src="../images/img-top-user-2.png" />
    				</div>
    			</div>
    			<header>
	    			<div>

	    			</div>
    			</header>
    		</div>
    		);
    }
}
