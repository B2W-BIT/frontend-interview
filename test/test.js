import React from 'react';
import renderer from 'react-test-renderer';

import Banner from '../src/components/Banner.jsx';
import Tweet from '../src/components/Tweet.jsx';
import Header from '../src/components/Header.jsx';
import TwitterUser from '../src/components/TwitterUser.jsx';
import User from '../src/components/User.jsx';

it('renders Banner', () => {
  const banner1 = renderer.create(<Banner src="https://pbs.twimg.com/profile_banners/31472986/1390867474/600x200" alt="banner" />).toJSON();
  expect(banner1).toMatchSnapshot();


  const banner3 = renderer.create(<Banner />).toJSON();
  expect(banner3).toMatchSnapshot();
});

it('renders Tweet', () => {

  const tweet1 = renderer.create(<Tweet text="@T_Lorena, a gente pode te ajudar por aqui. Fala com a gente por DM e conta o que aconteceu!" in_reply_to_screen_name="T_Lorena" user={{
      "id": 35019751,
      "id_str": "35019751",
      "name": "americanas.com",
      "screen_name": "americanascom",
      "location": "Brasil",
      "description": "A maior loja na palma da sua mão: https://t.co/J5pdYz5brU",
      "url": "http://t.co/PfJKJTZmj1",
      "entities": {
        "url": {
          "urls": [
            {
              "url": "http://t.co/PfJKJTZmj1",
              "expanded_url": "http://www.americanas.com",
              "display_url": "americanas.com",
              "indices": [
                0,
                22
              ]
            }
          ]
        },
        "description": {
          "urls": [
            {
              "url": "https://t.co/J5pdYz5brU",
              "expanded_url": "https://goo.gl/Ntg9wN",
              "display_url": "goo.gl/Ntg9wN",
              "indices": [
                34,
                57
              ]
            }
          ]
        }
      },
      "protected": false,
      "followers_count": 246609,
      "friends_count": 2415,
      "listed_count": 2547,
      "created_at": "Fri Apr 24 19:51:14 +0000 2009",
      "favourites_count": 31,
      "utc_offset": -10800,
      "time_zone": "Brasilia",
      "geo_enabled": false,
      "verified": true,
      "statuses_count": 26590,
      "lang": "en",
      "contributors_enabled": false,
      "is_translator": false,
      "is_translation_enabled": false,
      "profile_background_color": "E60014",
      "profile_background_image_url": "http://pbs.twimg.com/profile_background_images/754180801/ea96f827e4c0f1f5f368dff6bc1f545e.gif",
      "profile_background_image_url_https": "https://pbs.twimg.com/profile_background_images/754180801/ea96f827e4c0f1f5f368dff6bc1f545e.gif",
      "profile_background_tile": true,
      "profile_image_url": "http://pbs.twimg.com/profile_images/777959576482287617/taEJPem9_normal.jpg",
      "profile_image_url_https": "https://pbs.twimg.com/profile_images/777959576482287617/taEJPem9_normal.jpg",
      "profile_banner_url": "https://pbs.twimg.com/profile_banners/35019751/1474315029",
      "profile_link_color": "A8A1A2",
      "profile_sidebar_border_color": "FFFFFF",
      "profile_sidebar_fill_color": "F5FDFF",
      "profile_text_color": "333333",
      "profile_use_background_image": true,
      "has_extended_profile": false,
      "default_profile": false,
      "default_profile_image": false,
      "following": null,
      "follow_request_sent": null,
      "notifications": null,
      "translator_type": "none"
    }} entities={{
      "hashtags": [],
      "symbols": [],
      "user_mentions": [
        {
          "screen_name": "T_Lorena",
          "name": "Lorena Lima",
          "id": 58021583,
          "id_str": "58021583",
          "indices": [
            0,
            9
          ]
        }
      ],
      "urls": []
    }} />).toJSON();
  expect(tweet1).toMatchSnapshot();


  const tweet3 = renderer.create(<Tweet user={{}} entities={{}} />).toJSON();
  expect(tweet3).toMatchSnapshot();
});

it('renders Header', () => {
  const header1 = renderer.create(<Header tweets={ 5 } following={ 10 } followers={ 15 } likes={ 20 } />).toJSON();
  expect(header1).toMatchSnapshot();


  const header3 = renderer.create(<Header />).toJSON();
  expect(header3).toMatchSnapshot();
});

it('renders TwitterUser', () => {
  const twitteruser1 = renderer.create(<TwitterUser verified={ true } name="rafael" screen_name="rafacesar" />).toJSON();
  expect(twitteruser1).toMatchSnapshot();


  const twitteruser3 = renderer.create(<TwitterUser />).toJSON();
  expect(twitteruser3).toMatchSnapshot();

  const twitteruser4 = renderer.create(<TwitterUser verified={ false } name="rafael" screen_name="rafacesar" />).toJSON();
  expect(twitteruser4).toMatchSnapshot();
});

it('renders User', () => {


  const user1 = renderer.create(<User img="https://pbs.twimg.com/profile_banners/31472986/1390867474/600x200" verified={ true } name="rafael" screen_name="rafacesar" description="A maior loja na palma da sua mão: https://t.co/J5pdYz5brU" entities={ {
    "url": {
      "urls": [
        {
          "url": "http://t.co/PfJKJTZmj1",
          "expanded_url": "http://www.americanas.com",
          "display_url": "americanas.com",
          "indices": [
            0,
            22
          ]
        }
      ]
    },
    "description": {
      "urls": [
        {
          "url": "https://t.co/J5pdYz5brU",
          "expanded_url": "https://goo.gl/Ntg9wN",
          "display_url": "goo.gl/Ntg9wN",
          "indices": [
            34,
            57
          ]
        }
      ]
    }
  } } location="Brasil" url="https://twitter.com" since="1490660405657" />).toJSON();
  expect(user1).toMatchSnapshot();


  const user3 = renderer.create(<User />).toJSON();
  expect(user3).toMatchSnapshot();
});