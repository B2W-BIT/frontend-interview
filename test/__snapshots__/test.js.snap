// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Banner 1`] = `
<img
  alt="banner"
  className="img-responsive"
  src="https://pbs.twimg.com/profile_banners/31472986/1390867474/600x200"
/>
`;

exports[`renders Banner 2`] = `
<img
  alt=""
  className="img-responsive"
  src=""
/>
`;

exports[`renders Header 1`] = `
<div
  className="header-info"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-sm-offset-3 col-md-offset-3 col-lg-offset-3 col-xs-6 col-sm-2 col-md-2 col-lg-1"
      >
        Tweets 
        <br />
         
        5
      </div>
      <div
        className="col-xs-6 col-sm-2 col-md-2 col-lg-1"
      >
        Seguindo 
        <br />
         
        10
      </div>
      <div
        className="col-xs-6 col-sm-2 col-md-2 col-lg-1"
      >
        Seguidores 
        <br />
         
        15
      </div>
      <div
        className="col-xs-6 col-sm-2 col-md-2 col-lg-1"
      >
        Curtidas 
        <br />
         
        20
      </div>
    </div>
  </div>
</div>
`;

exports[`renders Header 2`] = `
<div
  className="header-info"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-sm-offset-3 col-md-offset-3 col-lg-offset-3 col-xs-6 col-sm-2 col-md-2 col-lg-1"
      >
        Tweets 
        <br />
         
        0
      </div>
      <div
        className="col-xs-6 col-sm-2 col-md-2 col-lg-1"
      >
        Seguindo 
        <br />
         
        0
      </div>
      <div
        className="col-xs-6 col-sm-2 col-md-2 col-lg-1"
      >
        Seguidores 
        <br />
         
        0
      </div>
      <div
        className="col-xs-6 col-sm-2 col-md-2 col-lg-1"
      >
        Curtidas 
        <br />
         
        0
      </div>
    </div>
  </div>
</div>
`;

exports[`renders Tweet 1`] = `
<div
  className="tweet-block row"
>
  <div
    className="col-xs-offset-1 col-xs-11"
  >
    Em resposta a 
    Lorena Lima
  </div>
  <div
    className="col-xs-1 tweet-img"
  >
    <img
      alt=""
      className="img-responsive"
      src="https://pbs.twimg.com/profile_images/777959576482287617/taEJPem9_normal.jpg"
    />
  </div>
  <div
    className="col-xs-11"
  >
    <span>
      <span
        className="title verified"
      >
        americanas.com
      </span>
      <span
        className="accountid"
      >
        @
        americanascom
      </span>
    </span>
    - 
    Invalid date
    <br />
    <a
      href="http://twitter.com/T_Lorena"
      style={Object {}}
    >
      @Lorena Lima
    </a>
    , a gente pode te ajudar por aqui. Fala com a gente por DM e conta o que aconteceu!
  </div>
</div>
`;

exports[`renders Tweet 2`] = `
<div
  className="tweet-block row"
>
  <div
    className="col-xs-1 tweet-img"
  >
    <img
      alt=""
      className="img-responsive"
      src={undefined}
    />
  </div>
  <div
    className="col-xs-11"
  >
    <span>
      <span
        className="title"
      >
        
      </span>
      <span
        className="accountid"
      >
        @
        
      </span>
    </span>
    - 
    Invalid date
    <br />
  </div>
</div>
`;

exports[`renders TwitterUser 1`] = `
<span>
  <span
    className="title verified"
  >
    rafael
  </span>
  <span
    className="accountid"
  >
    @
    rafacesar
  </span>
</span>
`;

exports[`renders TwitterUser 2`] = `
<span>
  <span
    className="title"
  >
    
  </span>
  <span
    className="accountid"
  >
    @
    
  </span>
</span>
`;

exports[`renders TwitterUser 3`] = `
<span>
  <span
    className="title"
  >
    rafael
  </span>
  <span
    className="accountid"
  >
    @
    rafacesar
  </span>
</span>
`;

exports[`renders User 1`] = `
<div
  className="user-info"
>
  <img
    alt="rafael"
    className="center-block"
    src="https://pbs.twimg.com/profile_banners/31472986/1390867474/600x200"
  />
  <h1>
    <span>
      <span
        className="title verified"
      >
        rafael
      </span>
      <span
        className="accountid"
      >
        @
        rafacesar
      </span>
    </span>
  </h1>
  <p>
    A maior loja na palma da sua m√£o: 
    <a
      href="https://goo.gl/Ntg9wN"
      style={Object {}}
    >
      goo.gl/Ntg9wN
    </a>
  </p>
  <p
    className="location"
  >
    Brasil
  </p>
  <p
    className="url"
  >
    https://twitter.com
  </p>
  <p
    className="since"
  >
    Participa desde 
    Invalid date
  </p>
</div>
`;

exports[`renders User 2`] = `
<div
  className="user-info"
>
  <img
    alt=""
    className="center-block"
    src=""
  />
  <h1>
    <span>
      <span
        className="title"
      >
        
      </span>
      <span
        className="accountid"
      >
        @
        
      </span>
    </span>
  </h1>
  <p>
    
  </p>
  <p
    className="location"
  >
    
  </p>
  <p
    className="url"
  >
    
  </p>
  <p
    className="since"
  >
    Participa desde 
    Invalid date
  </p>
</div>
`;
